test/open_api_typesense_connection_test.exs

```diff
@@ -0,0 +1,42 @@
+defmodule OpenApiTypesense.ConnectionTest do
+  use ExUnit.Case, async: true
+
+  alias OpenApiTypesense.Connection
+
+  test "new/0 using the default config to creates a connection struct" do
+    conn = Connection.new()
+
+    assert conn == %Connection{
+             api_key: "xyz",
+             host: "localhost",
+             port: 8108,
+             scheme: "http"
+           }
+  end
+
+  test "new/1 with custom fields creates a connection struct" do
+    conn =
+      Connection.new(%{
+        host: "otherhost",
+        port: 9200,
+        scheme: "https",
+        api_key: "myapikey"
+      })
+
+    assert conn == %Connection{
+             api_key: "myapikey",
+             host: "otherhost",
+             port: 9200,
+             scheme: "https"
+           }
+  end
+
+  test "new/1 with empty map raises ArgumentError" do
+    msg = "Missing required fields: [:port, :scheme, :host, :api_key]"
+    assert_raise ArgumentError, msg, fn -> Connection.new(%{}) end
+  end
+
+  test "new/1 with invalid data type raises ArgumentError" do
+    assert_raise ArgumentError, fn -> Connection.new("invalid") end
+  end
+end
```

Comment:
```
$ mix test --cover
Running ExUnit with seed: 952147, max_cases: 24

......................................................................
Finished in 1.6 seconds (1.5s async, 0.08s sync)
70 tests, 0 failures
----------------
COV    FILE                                        LINES RELEVANT   MISSED
 88.5% lib/open_api_typesense/client.ex              197       35        4
100.0% lib/open_api_typesense/connection.ex           78        8        0
100.0% lib/open_api_typesense/operations/analyt      213       21        0
100.0% lib/open_api_typesense/operations/collec      297       33        0
100.0% lib/open_api_typesense/operations/conver      168       18        0
100.0% lib/open_api_typesense/operations/curati      115       13        0
100.0% lib/open_api_typesense/operations/debug.       43        4        0
100.0% lib/open_api_typesense/operations/docume      511       66        0
100.0% lib/open_api_typesense/operations/health       33        3        0
100.0% lib/open_api_typesense/operations/keys.e      119       14        0
100.0% lib/open_api_typesense/operations/operat      110       12        0
100.0% lib/open_api_typesense/operations/overri       38        4        0
100.0% lib/open_api_typesense/operations/preset      135       15        0
100.0% lib/open_api_typesense/operations/stopwo      147       16        0
100.0% lib/open_api_typesense/operations/synony      138       16        0
[TOTAL]  98.5%
----------------
```

Probably is a good idea to add tests to `lib/open_api_typesense/client.ex`? Good jobs on adding moar tests to it!